### PR TITLE
BigQuery Job Updaters and EncryptionConfiguration

### PR DESCRIPTION
**Pull request by quartzmo**
This PR merges into master the `bigquery` topic branch that collected the changes from #1973, #2003, #2014, #2015 and #2016.

### CODE REVIEW

**Comment by quartzmo** [comment]
@tswast @keylimetoolbox There were a number of conflicts between #1973 and #2000 that I needed to manually resolve while rebasing this branch on master. All tests are passing (including acceptance tests), but PTAL just to be sure the Load Job code still looks good.

File: google-cloud-bigquery/lib/google/cloud/bigquery/table.rb [hunk 1]
```diff
@@ -1546,18 +1613,28 @@ def load_job files, format: nil, create: nil, write: nil,
...
+                                     max_bad_records: max_bad_records,
+                                     quote: quote, skip_leading: skip_leading,
+                                     dryrun: dryrun, schema: schema,
+                                     labels: labels, autodetect: autodetect,
+                                     null_marker: null_marker
+
+          yield updater if block_given?
+
+          job_gapi = updater.to_gapi
+          if local_file? files
```
Comment (tswast) [inline google-cloud-bigquery/lib/google/cloud/bigquery/table.rb]: I assume `local_file?` is okay getting a list now?

Comment (blowmage) [reply google-cloud-bigquery/lib/google/cloud/bigquery/table.rb]: No, only one local file can be provided. An array of URLs can be provided, either as strings or URI objects or storage file objects.

Comment (tswast) [reply google-cloud-bigquery/lib/google/cloud/bigquery/table.rb]: Does it return false when given array? That sounds like it should be the correct behavior.

Comment (quartzmo) [reply google-cloud-bigquery/lib/google/cloud/bigquery/table.rb]: @tswast I struggled to understand this as well while doing the rebase.

I don't think the docs make it clear that only one local file can be provided:

> A file or the URI of a Google Cloud Storage file, or an Array of  those, containing data to load into the table.

However, I'm fairly confident that in this case I preserved the original behavior of #2000 in my manual conflict resolution.

Comment (blowmage) [reply google-cloud-bigquery/lib/google/cloud/bigquery/table.rb]: Yes, `local_file?` will return `false` if passed an array.

Comment (jeremywadsack) [reply google-cloud-bigquery/lib/google/cloud/bigquery/table.rb]: Maybe a doc update if this is confusing?

Also , why not use `load_local_or_uri` here?

**Comment by jeremywadsack** [comment]
Looks like this preserves the changes in 2000.

**Pull request merged**